runtime/compiler/control/CompilationThread.cpp

```diff
@@ -8131,14 +8128,18 @@ TR::CompilationInfoPerThreadBase::wrappedCompile(J9PortLibrary *portLib, void *
 
 #if defined(J9VM_OPT_JITSERVER)
          // If the options come from a remote party, skip the setup options process
-         if (that->_methodBeingCompiled->_clientOptions != NULL)
-            {
-            TR_ASSERT(that->_methodBeingCompiled->isOutOfProcessCompReq(), "Options are already provided only for JITServer");
-            options = TR::Options::unpackOptions(that->_methodBeingCompiled->_clientOptions, that->_methodBeingCompiled->_clientOptionsSize, that, vm, p->trMemory());
-            options->setLogFileForClientOptions();
-            // The following is a hack to prevent the JITServer from allocating
-            // a sentinel entry for the list of runtime assumptions kept in the compiler object
-            options->setOption(TR_DisableFastAssumptionReclamation);
+
+         if (that->_methodBeingCompiled->isOutOfProcessCompReq())
+            {
+            auto compInfoPTRemote = static_cast<TR::CompilationInfoPerThreadRemote *>(that);
+            if (compInfoPTRemote->getClientOptions())
```

Comment:
The out-of-process compilation needs the options in order to continue.
I would replace the `if` statement with a fatal assert that the options must exist.

Reply:
Done.